R/cloud.R

```diff
@@ -530,15 +530,20 @@ cloud_email <- function(type = c("broken", "failed"), job_id = cloud_job(), pkg
 #' This is automatically set by [cloud_check()] and only lasts for the current R session.
 #' @param job_id If not `NULL`, sets the active `job_id` to the input.
 #' @export
-cloud_job <- function(job_id = NULL) {
-  if (!is.null(job_id)) {
-    cloud_data$job_id <- job_id
+cloud_job <- function(pkg = ".", job_id = NULL) {
+  cloud_data$job_id <- job_id %||% cloud_data$job_id
+  if (!is.null(cloud_data$job_id)) {
+    return(invisible(cloud_data$job_id))
   }
 
-  if (is.null(cloud_data$job_id)) {
-    stop("No current job, please specify the `job_id` explicitly, or run a job with `cloud_check()`", call. = FALSE)
+  pkg <- pkg_check(pkg)
+  cloud_dirs <- list.dirs(dir_find(pkg, "cloud"), recursive = FALSE)
+  if (length(cloud_dirs) < 1) {
+    stop("No previous `cloud_check()` results found, can't discover `job_id`", call. = FALSE)
   }
-
+  latest <- cloud_dirs[which.max(file.info(cloud_dirs)$mtime)]
+  cloud_data$job_id <- basename(latest)
+  cli_alert_success("Most recent batch job {.arg job_id}: {.val {cloud_data$job_id}}")
```

Comment:
Does this message and the error just above look like what we'd want?

Reply:
Maybe `Most recent cloud job`?